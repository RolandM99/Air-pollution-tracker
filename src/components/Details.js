/* eslint-disable react/prop-types */ // TODO: upgrade to latest eslint tooling
import React from 'react';
import { useSelector } from 'react-redux';
import {
  WiDaySunny, WiDayThunderstorm, WiDaySnowWind, WiDayRain, WiCloudy, WiDayRainMix,
  WiDayHaze, WiDayFog, WiSmoke, WiTornado, WiCloudyGusts,
} from 'react-icons/wi';

const Details = () => {
  const city = useSelector((state) => state.dataReducer);
  const icon = () => {
    switch (city.main) {
      case 'Clear':
        return (<WiDaySunny />);
      case 'Thunderstorm':
        return (<WiDayThunderstorm />);
      case 'Snow':
        return (<WiDaySnowWind />);
      case 'Rain':
        return (<WiDayRain />);
      case 'Clouds':
        return (<WiCloudy />);
      case 'Drizzle':
        return (<WiDayRainMix />);
      case 'Mist':
        return (<WiDayFog />);
      case 'Smoke':
        return (<WiSmoke />);
      case 'Fog':
        return (<WiDayFog />);
      case 'Haze':
        return (<WiDayHaze />);
      case 'Dust':
        return (<WiDayRainMix />);
      case 'Sand':
        return (<WiCloudyGusts />);
      case 'Ash':
        return (<WiCloudyGusts />);
      case 'Squall':
        return (<WiTornado />);
      case 'Tornado':
        return (<WiTornado />);
      default:
        return (<div className="none" />);
    }
  };
  return (
    <div className="weather">
      <div className="weather-content">
        <h1 className="name">
          {city.name}
        </h1>
        <div className="icon">{icon()}</div>
        <h2 className="description">{city.description}</h2>
        <h3 className="temp">
          Temp :
          {city.temp}
          {'\u00b0'}
          F
        </h3>
        <div className="others">
          <span className="humidity">
            Humidity :
            {city.humidity}
          </span>
          <span className="pressure">
            Pressure :
            {city.pressure}
          </span>
        </div>
      </div>
    </div>
  );
};

export default Details;
