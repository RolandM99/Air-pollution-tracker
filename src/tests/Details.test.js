import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Details from '../components/Details';

describe('Details Component test', () => {
  it('Renders the details of city weather data', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Details />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
