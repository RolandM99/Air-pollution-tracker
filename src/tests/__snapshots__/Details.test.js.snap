// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component test Renders the details of city weather data 1`] = `
<div
  className="text-center"
>
  <p>
    loading...
  </p>
</div>
`;
